packages/atomic/src/components/common/result-list/result-list-common-interface.ts

```diff
@@ -0,0 +1,57 @@
+import {AnyBindings} from '../interface/bindings';
+import {
+  ResultDisplayDensity,
+  ResultDisplayImageSize,
+  ResultDisplayLayout,
+} from '../layout/display-options';
+import {AnyResult} from '../interface/result';
+import {ResultTemplate} from '@coveo/headless';
+import {FocusTargetController} from '../../../utils/accessibility-utils';
+
+export interface ResultListRenderer {
+  getTemplateContent(result: AnyResult): HTMLElement | DocumentFragment;
+  getResultId(result: AnyResult): string;
+  listClasses: string;
+  setNewResultRef(element: HTMLElement, resultIndex: number): void;
+}
+
+export interface ResultListCommonProps {
+  bindings: AnyBindings;
+  host: HTMLElement;
+  loadingFlag: string;
+  getDisplay(): ResultDisplayLayout;
```

Comment:
~Why are these all `getX()` function instead of just the value directly ?~

nvm: your explanation is below.